Makefile

```diff
@@ -11,6 +11,24 @@ CFLAGS = -std=gnu99 -O2 -Wall -Wextra
 CFLAGS += -Wno-unused-label
 CFLAGS += -include src/common.h
 
+# In the system test suite, the executable is an ELF file.
+# However, the Linux kernel emulation includes the Image, DT, and
+# root filesystem (rootfs). Therefore, the test suite needs this
+# flag to load the ELF and differentiate it from the kernel emulation.
+ENABLE_USE_ELF ?= 0
+ifeq ($(call has, USE_ELF), 1)
+DEFINE_USE_ELF := -DUSE_ELF
+endif
+
+# The MMU test suite relies on rv32emu's syscall function instead of
+# its syscall table, unlike the Linux kernel. Therefore, use the
+# ON_TEST macro to distinguish between the Linux kernel and MMU test
+# suite as the emulation target.
+ENABLE_ON_TEST ?= 0
+ifeq ($(call has, ON_TEST), 1)
+DEFINE_ON_TEST := -DON_TEST
```

Comment:
Ditto. Use `call set-feature` explicitly and ensure the configuration dependency.